### DISCUSSION TITLE
Unable to boot using iPXE becouse link get failed during DHCP

### DISCUSSION THREAD

**Discussion by maestrx** [ipxe/ipxe#1046] [Q&A]
I'm not able to boot a server using iPXE. Other boot methods are working. Issue is that at the moment iPXE is trying to get IP using DHCP, the link goes down(logs on the switch side). It seems that iPXE is causing the link to go down but Im not able to find the root cause. Any idea?

Thanks

**Comment by NiKiZe** [comment]
Errors from iPXE always have a message including a url to ipxe.org, please include this message and URL. That URL should also provide some common causes and fixes.

In this case, check STP settings?
Also depending on which iPXE binary you are using, native drivers for the NIC might be used, this includes the link being reset.

**Reply by maestrx** [reply]
Links:
https://ipxe.org/err/040ee1
https://ipxe.org/err/4c1060

![obrazek](https://github.com/ipxe/ipxe/assets/6150459/2c2bb18f-a85c-4091-83c6-77be9cb97648)

![obrazek](https://github.com/ipxe/ipxe/assets/6150459/7c621aa2-27d4-48df-9319-ab794c5e3e55)

I dont see any immediate hint on what Im doing wrong. Link is up and goes down only at the moment iPXE is trying to get IP. Crazy thing is that I can boot using the same interface live Alma linux (not using iPXE) but iPXE is failing... Using exactly identical environment. 

Any idea how to debug this?

Thanks

Switch port config:
interface Ethernet1/67
  switchport
  switchport access vlan 230
  spanning-tree port type edge
  spanning-tree bpduguard enable
  no shutdown

**Reply by NiKiZe** [reply]
First link you posted have several suggestions, last line in that page even has suggested Cisco switch config for STP, before that it asks you to run ifstat to collect more data, and it would also help in this case to see which driver iPXE uses for your NIC.

**Reply by maestrx** [reply]
![obrazek](https://github.com/ipxe/ipxe/assets/6150459/46c43f7d-7e85-447a-87a6-32b84a9fd8ff)

This shows that server is able to boot over network, but iPXE later fails on doing the same...

**Reply by maestrx** [reply]
STP configuration of the interface is in the message above. ifstat shows TX but no RX. 

Related network adapter is: "HPE FlxFbrc 10Gb 4p 536FLR-T Adptr"

Not sure how to check what drivers are bundled in the iPXE we are booting. Its provided as part of the solution from DRP (RackN)...

**Reply by NiKiZe** [reply]
The screenshot shows that it is using NII driver. In terms of RX and RXE, that data is interesting after dhcp has been tried and fail.

In terms of "shows server able to boot" noone has claimed otherwise, but it is a bit different, first, how long did it take?

Please go to the iPXE shell, Run ifopen, then wait 60 seconds and run dhcp.

Please read that error page once more, cerfully. Don't jump to conclusions about what is going on, actually check each step it suggests.
